### ISSUE TITLE
Some Unittest Layer input is uninitialized but pass test

### ISSUE THREAD

**Issue by zhoonit** [nnstreamer/nntrainer#1172]
Some weight loading logic is inside `reinitialize` but it is not called for some forwarding tests after https://github.com/nnstreamer/nntrainer/pull/1001/files. 

This leaves additional mystery how those tests managed to pass so far.

<details>

```cpp
TEST_F(nntrainer_BatchNormalizationLayer, forward_backward_training_01_p) {
  layer.setTrainable(true);
  sharedConstTensor forward_result;
  //////////////////////////// actually reinitialize() should be called here

  allocateMemory();
  EXPECT_NO_THROW(forward_result =
                    layer.forwarding_with_val({MAKE_SHARED_TENSOR(in)})[0]);
  matchOutput(*forward_result, "tc_bn_fc_1_goldenBNResultForward.out");

  nntrainer::Tensor backward_in(layer.getOutputDimension()[0]);
  loadFile("tc_bn_fc_1_goldenBNLayerBackwardDxIn.out", backward_in);

  nntrainer::Tensor backward_result =
    *layer.backwarding_with_val(1, {MAKE_SHARED_TENSOR(backward_in)}, opt)[0];

  matchOutput(backward_result, "tc_bn_fc_1_goldenBNLayerBackwardDx.out");
}
```

`in` is loaded at `reinitialize` but it is not called
</details>

**Comment by kparichay** [comment]
@zhoonit Please check, #1181 might have resolved this.

**Comment by zhoonit** [comment]
Yes this is resolved with #1181. Thanks
